crates/storage/provider/src/providers/historical.rs

```diff
@@ -64,6 +64,24 @@ impl<'a, 'b, TX: DbTx<'a>> AccountProvider for HistoricalStateProviderRef<'a, 'b
             Ok(self.tx.get::<tables::PlainAccountState>(address)?)
         }
     }
+
+    /// Get account code by its address
+    fn account_code(&self, addr: Address) -> Result<Option<Bytes>> {
+        // Get basic account information
+        let acc = match self.basic_account(addr)? {
+            Some(acc) => acc,
+            None => return Ok(None),
+        };
+        // Get code hash
+        let code_hash = acc.bytecode_hash;
+        // Get the code from the code hash
+        self.code_by_hash(code_hash.unwrap())
+    }
+
+    /// Get account code by its hash
+    fn code_by_hash(&self, code_hash: H256) -> Result<Option<Bytes>> {
```

Comment:
we already have this in `StateProvider`

and since `StateProvider: AccountProvider` we can move the `fn account_code` to the `StateProvider` trait instead.
then we can use the available `bytecode_by_hash` function